File: 03-frontend/angular-ecommerce/src/app/common/purchase.ts
import { Address } from './address';
import { Customer } from './customer';
import { Order } from './order';
import { OrderItem } from './order-item';

// Use this Object to send all data that received from checkout form
export class Purchase {
    customer!: Customer;
    shippingAddress!: Address;
    billingAddress!: Address;
    order!: Order;
    orderItems!: OrderItem[];
}

File: 02-backend/spring-boot-ecommerce/src/main/java/com/simplilearn/ecommerce/dto/Purchase.java
package com.simplilearn.ecommerce.dto;

import com.simplilearn.ecommerce.entity.Address;
import com.simplilearn.ecommerce.entity.Customer;
import com.simplilearn.ecommerce.entity.OrderItem;
import com.simplilearn.ecommerce.entity.Order;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customers;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;


}
